crates/bevy_audio/src/audio_source.rs

```diff
@@ -74,3 +160,11 @@ impl Decodable for AudioSource {
         rodio::Decoder::new(Cursor::new(self.clone())).unwrap()
     }
 }
+
+/// A trait that allows adding a custom audio source to the object. This is implemented for [`App`][bevy_app::App] to allow registering custom [`Decodable`] types.
+pub trait AddAudioSource {
+    /// Registers an audio source. The type must implement [`Decodable`], so that it can be converted to [`rodio::Source`], and [`Asset`], so that it can be registered. To use this method on [`App`][bevy_app::App] the [`Audio`][super::AudioPlugin] [`Asset`][bevy_asset::AssetPlugin] plugins must be added to the app.    
```

Comment:
```suggestion
    /// Registers an audio source.
    /// The type must implement [`Decodable`],
    /// so that it can be converted to a [`rodio::Source`] type,
    /// and [`Asset`], so that it can be registered as an asset.
    /// To use this method on the [`App`][bevy_app::App],
    /// the [audio][super::AudioPlugin] and [asset][bevy_asset::AssetPlugin] plugins must be added first.    
```